### PR TITLE
[workflow] Deprecate workflow.create

### PR DESCRIPTION
**Pull request by suquark**
## Why are these changes needed?

After deprecating `Workflow` class in #25618 , `workflow.create` becomes a redundant API and should be deprecated.

We update examples and docs correspondingly.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/workflows/basics.rst [hunk 1]
```diff
@@ -297,15 +296,50 @@ Note that tasks that have side-effects still need to be idempotent. This is beca
...
+    def factorial(n: int) -> int:
+        if n == 1:
+            return 1
+        else:
+            return multiply(n, factorial.bind(n - 1))
+
+    @ray.remote
+    def multiply(a: int, b: int) -> int:
+        return a * b
+
```
Comment (iycheng) [inline doc/source/workflows/basics.rst:340]: Is the syntax correct?

File: doc/source/workflows/comparison.rst [hunk 2]
```diff
@@ -7,7 +7,10 @@ Workflows is built on top of Ray, and offers a mostly consistent subset of its A
 
 ``func.remote`` vs ``func.bind``
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-With Ray tasks, ``func.remote`` will submit a remote task to run eagerly. In Ray workflows, ``func.bind`` is used to create a DAG, and the DAG is converted into a workflow. Execution of the workflow is deferred until ``.run(workflow_id="id")`` or ``.run_async(workflow_id="id")`` is called on the ``Workflow``. Specifying the workflow id allows for resuming of the workflow by its id in case of cluster failure.
+With Ray tasks, ``func.remote`` will submit a remote task to run eagerly.
+In Ray workflows, ``func.bind`` is used to create a DAG, and the DAG is converted into a workflow.
```
Comment (iycheng) [inline doc/source/workflows/comparison.rst:11]: DAG vs remote task seems not comparable? Maybe a node in the dag?

**Review by fishbone — COMMENTED** [review]
LG. Some small comments.

**Review by fishbone — APPROVED** [review]

**Comment by iycheng** [comment]
```
//python/ray/workflow:tests/test_basic_workflows_2                      TIMEOUT in 3 out of 3 in 900.0s
```

**Comment by suquark** [comment]
The failed CI test seems unrelated. I'll merge the PR.

**Pull request merged**